static/app/views/explore/contexts/spanTagsContext.tsx

```diff
@@ -21,20 +21,21 @@ export const SpanTagsContext = createContext<TypedSpanTags | undefined>(undefine
 interface SpanTagsProviderProps {
   children: React.ReactNode;
   dataset: DiscoverDatasets;
+  enabled?: boolean;
```

Comment:
This is not being passed from the explore pages right now and would break the tag fetching. Can we make this a required prop instead?

Reply:
So if `enabled` is `undefined`, then `enabled: dataset === DiscoverDatasets.SPANS_INDEXED && enabled,` actually also evaluates to `undefined` (if the dataset check returns true). Which means underlying fetch hooks will fall back to the default enabled behaviour. 

This behaviour isn't really obvious though when reading the code, so I agree we can update the `enabled` prop to be required!